### PR TITLE
Script for defect 2401

### PR DESCRIPTION
**Pull request by TMyza**
ATF Test Scripts to check [#2401](https://github.com/smartdevicelink/sdl_core/issues/2401)

Summary
A set of test scripts to verify that issue is fixed

This PR is **not ready** for review.
This is **draft** version of the script.

### CODE REVIEW

File: test_scripts/Defects/5_0/2401_SDL_behavior_in_case_mobile_app_sends_request_with_at_least_one_choiseSet_without_vrCommands_parameter_and_at_least_one_choiseSet_with_vrCommands_at_VR_ONLY_or_BOTH_mode.lua [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+local runner = require('user_modules/script_runner')
+local common = require('user_modules/sequences/actions')
+
+-- [[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+local paramsVrOnly = {
+    initialText = "StartPerformInteraction1",
+    interactionMode = "VR_ONLY",
+    interactionChoiceSetIDList = { 10 },
```
Comment (aderiabin) [inline test_scripts/Defects/5_0/2401_SDL_behavior_in_case_mobile_app_sends_request_with_at_least_one_choiseSet_without_vrCommands_parameter_and_at_least_one_choiseSet_with_vrCommands_at_VR_ONLY_or_BOTH_mode.lua]: @TMyza It is existing an overriding of table field `interactionChoiceSetIDList` value here: initially this field has value `{ 10 }` and it is overridden with value `{ 11 }` in next line. 
Please fix this.

The same issue appears in `paramsBoth` table also.

File: test_scripts/Defects/5_0/2401_SDL_behavior_in_case_mobile_app_sends_request_with_at_least_one_choiseSet_without_vrCommands_parameter_and_at_least_one_choiseSet_with_vrCommands_at_VR_ONLY_or_BOTH_mode.lua [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        interactionChoiceSetID = 10,
+        choiceSet = {
+          {
+            choiceID = 10,
+            menuName = "Choice10"
+          }
+        } 
+    }
+    local cid = common.getMobileSession():SendRPC("CreateInteractionChoiceSet", choiceParams)
+    common.getHMIConnection():ExpectRequest("VR.PerformInteraction")
```
Comment (aderiabin) [inline test_scripts/Defects/5_0/2401_SDL_behavior_in_case_mobile_app_sends_request_with_at_least_one_choiseSet_without_vrCommands_parameter_and_at_least_one_choiseSet_with_vrCommands_at_VR_ONLY_or_BOTH_mode.lua]: @TMyza Why do you check HMI request `VR.PerformInteraction` and do not check HMI request `VR.AddCommand` here?

File: test_scripts/Defects/5_0/2401_SDL_behavior_in_case_mobile_app_sends_request_with_at_least_one_choiseSet_without_vrCommands_parameter_and_at_least_one_choiseSet_with_vrCommands_at_VR_ONLY_or_BOTH_mode.lua [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    initialText = "StartPerformInteraction1",
+    interactionMode = "VR_ONLY",
+    interactionChoiceSetIDList = { 10 },
+    interactionChoiceSetIDList = { 11 }
+}
+
+local paramsBoth = {
+    initialText = "StartPerformInteraction2",
+    interactionMode = "BOTH",
+    interactionChoiceSetIDList = { 10 },
```
Comment (aderiabin) [inline test_scripts/Defects/5_0/2401_SDL_behavior_in_case_mobile_app_sends_request_with_at_least_one_choiseSet_without_vrCommands_parameter_and_at_least_one_choiseSet_with_vrCommands_at_VR_ONLY_or_BOTH_mode.lua]: @TMyza Please avoid to create hardcode dependencies between variables with different visibility scopes.
Example of mentioned dependency from current script:
```
(local scope)paramsBoth.interactionChoiceSetIDList <==> (function createInteractionChoiceSetVR scope)choiceParams.interactionChoiceSetID
```

**Comment by dboltovskyi** [comment]
Livio team please do not review this PR for now since it hasn't passed our internal review yet.
Review and appropriate updates going be done till 1st of November.
Thank you.

**Comment by dboltovskyi** [comment]
The internal review iteration has been moved to the LuxoftSDL.

**Pull request closed without merging**